string.c

```diff
@@ -11525,6 +11525,189 @@ rb_enc_interned_str_cstr(const char *ptr, rb_encoding *enc)
+ *  - #index:: Returns the index of the _last_ occurrence of a given substring;
+ *             returns +nil+ if none found.
+ *  - #start_with?:: Returns +true+ if the string begins with any of the given substrings.
+ *  - #sum:: Returns a basic checksum for the string.
+ *  - #unicode_normalized?:: Returns +true+ if the string is in Unicode normalized form; +false+ otherwise.
+ *  - #valid_encoding?:: Returns +true+ if the string contains only characters that are valid
+ *                       for its encoding.
+ *
+ *  === Methods for Comparing
+ *
+ *  - #<=>::
+ *  - #==::
+ *  - #===::
+ *  - #casecmp::
+ *  - #casecmp?:: Returns +true+ if the string is equal to a given string after Unicode case folding;
+ *                +false+ otherwise.
+ *
+ *  === Methods for Fetching
+ *
+ *  - #[], #slice::
+ *  - #byteslice::
+ *  - #chr::
+ *  - #getbyte::
+ *
+ *  === Methods for Modifying
+ *
+ *  - #<<::
+ *  - #[]=::
+ *  - #capitalize!::
+ *  - #downcase!::
+ *  - #encode!::
+ *  - #gsub!::
+ *  - #force_encoding::
+ *  - #initialize_copy::
+ *  - #insert::
+ *  - #lstrip!::
+ *  - #next!::
+ *  - #replace::
+ *  - #reverse!::
+ *  - #rstrip!::
+ *  - #setbyte::
+ *  - #sub!::
+ *  - #succ!::
+ *  - #swapcase!::
+ *  - #tr!::
+ *  - #tr_s!::
+ *
+ *  === Methods for Deleting
+ *
+ *  - #chomp!::
+ *  - #chop!::
+ *  - #clear
+ *  - #delete::
+ *  - #delete_prefix::
+ *  - #delete_suffix::
+ *  - #lstrip::
+ *  - #rstrip::
+ *  - #scrub!::
+ *  - #slice!
+ *  - #squeeze!::
+ *  - #strip!::
+ *  - #unicode_normalize!::
+ *  - #upcase!::
+ *
+ *  === Methods for Iterating
+ *
+ *  - #each_byte::
+ *  - #each_char::
+ *  - #each_codepoint::
+ *  - #each_grapheme_cluster::
+ *  - #each_line::
+ *  - #upto::
+ *
+ *  === Methods for Converting
+ *
+ *  - #%::
+ *  - #*::
+ *  - #+::
+ *  - #bytes::
+ *  - #capitalize::
+ *  - #center::
+ *  - #chars::
+ *  - #chomp::
+ *  - #chop::
+ *  - #codepoints::
+ *  - #concat::
+ *  - #downcase::
+ *  - #dump::
+ *  - #encode::
+ *  - #gsub::
+ *  - #grapheme_clusters::
+ *  - #hex::
+ *  - #inspect::
+ *  - #intern::
+ *  - #lines::
+ *  - #ljust::
+ *  - #next::
+ *  - #oct::
+ *  - #ord::
+ *  - #partition::
+ *  - #prepend::
+ *  - #reverse::
+ *  - #rjust::
+ *  - #rpartition::
+ *  - #scan::
+ *  - #scrub::
+ *  - #split::
+ *  - #squeeze::
+ *  - #strip::
+ *  - #sub::
+ *  - #succ::
+ *  - #swapcase::
+ *  - #to_c::
+ *  - #to_d::
+ *  - #to_f::
+ *  - #to_i::
+ *  - #to_r::
+ *  - #to_s::
+ *  - #to_str::
+ *  - #to_sym::
+ *  - #toeuc::
+ *  - #tojis::
+ *  - #tolocale::
+ *  - #tosjis::
+ *  - #toutf16::
+ *  - #toutf32::
+ *  - #toutf8::
+ *  - #tr::
+ *  - #tr_s::
+ *  - #undump::
+ *  - #unicode_normalize::
+ *  - #unpack::
+ *  - #unpack1::
+ *  - #upcase::
+ *
+ *  === Methods for Encoding
+ *
+ *  - #b::
+ *  - #encode::
+ *  - #encode!::
+ *  - #encoding::
+ *  - #force_encoding::
+ *  - #valid_encoding?:: Returns +true+ if the string contains only characters that are valid
```

Comment:
This should be removed from either this section or the section above.

Reply:
I hadn't thought to keep a strict partition, but instead to dup when sensible.  Best example was the stuff we dumped from kconv.rb, some of which I'd duped as both encoding and query.

Reply:
Please pick the most appropriate section, so that a method is not listed more than once.

Reply:
Done.